editor/src/viewport_tools/tools/pen_tool.rs

```diff
@@ -310,16 +324,31 @@ fn add_to_curve(data: &mut PenToolData, input: &InputPreprocessorMessageHandler,
 		// Select the second to last segment's handle
 		data.shape_editor.set_shape_selected(0);
 		let handle_element = data.shape_editor.select_nth_anchor(0, -2);
-		handle_element.select_point(2, true, responses);
+		handle_element.select_point(ControlPointType::Handle2 as usize, true, responses);
 
 		// Select the last segment's anchor point
 		if let Some(last_anchor) = data.shape_editor.select_last_anchor() {
-			last_anchor.select_point(0, true, responses);
+			last_anchor.select_point(ControlPointType::Anchor as usize, true, responses);
 		}
 		data.shape_editor.set_selected_mirror_options(true, true);
 	}
 }
 
+// Replace a path element by ID
+fn replace_path_element(data: &mut PenToolData, transform: DAffine2, path_element_id: usize, replacement: PathEl, responses: &mut VecDeque<Message>) {
+	data.bez_path[path_element_id] = replacement;
+	if let Some(layer_path) = &data.path {
+		responses.push_back(apply_bez_path(layer_path.clone(), data.bez_path.clone(), transform));
+	}
+}
+
+// Remove a curve to the bez_path
```

Comment:
Comment doesn't really give me a clear understanding of what this does. Also these should all be doc comments.